### PR TITLE
[PROJ] update to 9.2.1, build with libtiff 4.5

### PR DESCRIPTION
**Pull request by visr**
See #6929

### CODE REVIEW

File: P/PROJ/build_tarballs.jl [hunk 1]
```diff
@@ -90,8 +87,8 @@ dependencies = [
     # Host SQLite needed to build proj.db
     HostBuildDependency("SQLite_jll")
     Dependency("SQLite_jll")
-    Dependency("Libtiff_jll")
-    Dependency("LibCURL_jll"; compat="7.73")
+    Dependency("Libtiff_jll"; compat="~4.5.1")
```
Comment (giordano) [inline P/PROJ/build_tarballs.jl:90]: I think this is potentially overly restrictive, they didn't break the ABI from 4.1 to 4.3, they just break it in an unpredictable way.

Also, as usual this stuff has to be fixed first in the registry

File: P/PROJ/build_tarballs.jl [hunk 2]
```diff
@@ -90,8 +87,8 @@ dependencies = [
...
+    Dependency("LibCURL_jll")
```
Comment (giordano) [inline P/PROJ/build_tarballs.jl:91]: Curl must have colpa bounds: https://github.com/JuliaPackaging/Yggdrasil/issues/3024

File: P/PROJ/build_tarballs.jl (hunk 1 above)
Comment (visr) [reply P/PROJ/build_tarballs.jl]: Ok, is `compat="4.5"` good? Done in cd47fa86b2f47ca0d51472bbd9b9dd2e0770384b

File: P/PROJ/build_tarballs.jl (hunk 2 above)
Comment (visr) [reply P/PROJ/build_tarballs.jl]: Should I leave it as `compat="7.73"`, or is the `compat="7.73,8"` I set in cd47fa86b2f47ca0d51472bbd9b9dd2e0770384b also ok?

Comment (giordano) [reply P/PROJ/build_tarballs.jl]: Well, you to allow v8 if you want to make this usable in newer versions

Comment (visr) [reply P/PROJ/build_tarballs.jl]: Doesn't `compat="7.73,8"` allow v8? Not sure what to put here.

Comment (giordano) [reply P/PROJ/build_tarballs.jl]: Yes, I was answering the previous question.

File: P/PROJ/build_tarballs.jl [hunk 3]
```diff
@@ -90,8 +87,8 @@ dependencies = [
     # Host SQLite needed to build proj.db
     HostBuildDependency("SQLite_jll")
     Dependency("SQLite_jll")
-    Dependency("Libtiff_jll")
-    Dependency("LibCURL_jll"; compat="7.73")
+    Dependency("Libtiff_jll"; compat="4.5")
```
Comment (visr) [inline P/PROJ/build_tarballs.jl:90]:
```suggestion
    Dependency("Libtiff_jll"; compat="4.5.1")
```
Apparently needed since 4.5.0 doesn't exist.

**Comment by giordano** [comment]
Delete system libcurl on x86_64 musl, we often have to do that

File: P/PROJ/build_tarballs.jl [hunk 4]
```diff
@@ -23,15 +24,18 @@ EXE_SQLITE3=${host_bindir}/sqlite3
...
     SQLITE3_LIBRARY=${libdir}/libsqlite3.${dlext}
     CURL_LIBRARY=${libdir}/libcurl.${dlext}
     TIFF_LIBRARY_RELEASE=${libdir}/libtiff.${dlext}
 fi
 
-if [[ "${target}" == x86_64-linux-musl* ]]; then
-    export LDFLAGS="-lcurl"
+if [[ ${target} == x86_64-linux-musl ]]; then
+    # ${libdir}/libcurl.so needs a libnghttp, and it prefers to load /usr/lib/libnghttp2.so for this.
+    # Unfortunately, that library is missing a symbol. Setting LD_LIBRARY_PATH is not enough to avoid this.
```
Comment (giordano) [inline P/PROJ/build_tarballs.jl:36]: `LD_LIBRARY_PATH` is only for dynamic loading, if has nothing to do with linking, it isn't surpring it doesn't help anything here.

**Pull request merged**